### ISSUE TITLE
Move F.pad into ATen

### ISSUE THREAD

**Issue by peterbell10** [pytorch/pytorch#73401]
## Issue description
`torch.nn.functional.pad` is currently implemented in python,
https://github.com/pytorch/pytorch/blob/abb55c53b3faf41fb1b2e9a69a1ca7b49c3e5b1b/torch/nn/functional.py#L4302

Which forces other functions to be defined in python, including `torch.stft`. Instead, the function should be moved to into `ATen`.

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345

**Comment by jbschlosser** [comment]
I love this. Removing python-side stuff like this also makes life easier wrt JIT FC (see #67565).

**Comment by ngimel** [comment]
There's a C++ implementation of torch::nn::functional::pad, but it's somewhat divergent from python now (at least for circular padding iirc)
